File: Inceptezpyspark-27/Sparkcore/Lab03_wholetextfile.py
from pyspark import SparkContext


def main():
    sc = SparkContext(master="local",appName="Lab03")
    sc.setLogLevel("ERROR")
    
    tdata = sc.wholeTextFiles("file:/home/hduser/sparkdata/sparkdata1")
    
    print(tdata.count())
        
    tdata.foreach(lambda x: print(x[0]))
    
    tdata.foreach(lambda x: print(x[1]))
     
    print("============================1")
    
    tdata1 = tdata.map(lambda x: x[0])
    
    filenames = tdata1.collect()
    
    print(type(filenames))
    
    print(filenames)
    
    lst = [10,20,30,"Apple"]
    
    rdd1 = sc.parallelize(lst)
    
    lst1 = rdd1.collect()
    
    print(lst1)

main()

File: InceptezSpark-27/src/inceptez/sparkcore/Lab03_wholetextfile.scala
package inceptez.sparkcore

import org.apache.spark.SparkContext


object Lab03_wholetextfile {
  
  def main(args:Array[String])=
   {
    val sc = new SparkContext(master="local",appName="Lab03")
    sc.setLogLevel("ERROR")
    
    val tdata = sc.wholeTextFiles("file:/home/hduser/sparkdata/sparkdata1")
    
    println(tdata.count())
        
    tdata.foreach(x => println(x._1))
    
    tdata.foreach(x => println(x._2))
     
    println("============================1")
    
    val tdata1 = tdata.map(x => x._1)
    
    val filenames = tdata1.collect()
    
    //val f = filenames.toList
    //println(f)
    
    //filenames.foreach(println)
    
    println(filenames.toList)
    
    val lst = List(10,20,30,"Apple")
    
    val rdd1 = sc.parallelize(lst)
    
    val lst1 = rdd1.collect()
    
    println(lst1.toList)
    
    
    
    
    
     
    
      
  }
}